src/sfizz/BeatClock.cpp

```diff
@@ -175,32 +181,35 @@ void BeatClock::fillBufferUpTo(unsigned delay)
     for (unsigned i = fillIdx; i < delay; ++i)
         beatsPerBarData[i] = sig.beatsPerBar;
 
-    if (!isPlaying_) {
-        if (fillIdx < delay) {
-            fill(absl::MakeSpan(&beatNumberData[fillIdx], delay - fillIdx), 0);
-            fill(absl::MakeSpan(&beatNumberPosition[fillIdx], delay - fillIdx), 0.0f);
-        }
-        currentCycleFill_ = fillIdx;
-        return;
-    }
-
     BBT clientPos = lastClientPos_;
-    const double beatsPerFrame = beatsPerSecond_ * samplePeriod_;
-
     const BBT hostPos = lastHostPos_;
     bool mustApplyHostPos = mustApplyHostPos_;
 
-    for (; fillIdx < delay; ++fillIdx) {
-        clientPos = BBT::fromBeats(sig, clientPos.toBeats(sig) + beatsPerFrame);
+    if (!isPlaying_) {
```

Comment:
Now if we're not playing, the beat position vector is set to the last known position *or* the host position if it was sent. Before it was initialized to zero. I think it makes more sense like this but I'll let you validate it.

Reply:
This implementation certainly makes sense.
For completeness, the host position should probably be also applied in the non-playing case.
It's imaginable that play head can be moved also while stopped.

Reply:
I think it is now, if you look below this line you have `clientPos = mustApplyHostPos ? hostPos : clientPos;`